### ISSUE TITLE
Doctrine TypeProvider does not work for class fields

### ISSUE THREAD

**Issue by maldoinc** [Haehnchen/idea-php-symfony2-plugin#1510]
Hi all! In the following example the type provider fails to recognize types when calling `getManager()` on doctrine. A call to the manager is often necessary as there might be multiple of them. 

Something to note is that on the following example all of the cases will work if one would use directly the `$registry` argument rather than the class field. Also the `RegistryInterface` is not in fact the default but rather a custom one inheriting `Symfony\Bridge\Doctrine\RegistryInterface` and overriding the default `@doctrine` service.

```
class SomeService
{
    /** @var RegistryInterface */
    private $registry;
    
    public function __construct(RegistryInterface $registry) {
        $this->registry = $registry;
        
        $works = $this->registry
            ->getRepository(Customer::class)
            ->find(1);

        $doesntWork = $this->registry
            ->getManager()
            ->getRepository(Customer::class)
            ->findBy(1);

        $neitherDoesThis = $this->registry
            ->getManager('main')
            ->getRepository(Customer::class)
            ->find(1);
    }
}
```

Additional info:

|Item|Version|
|---|---|
|PHP|7.3.13|
|Plugin|0.20.198|
|PhpStorm|Build #PS-201.7846.90, built on June 3, 2020|

**Comment by Jasonoro** [comment]
Same issue here. The interesting thing is that if you cast the manager to an `EntityManagerInterface` it does work:

```php
class TestController extends BaseAbstractController {

	public function doesntWork() {
		$em = $this->getDoctrine()->getManager();
		$customer = $em->getRepository(Customer::class)->find(1);
		// $customer type is \Doctrine\Persistence\ObjectRepository|null|object
	}

	public function doesWork() {
		/** @var EntityManagerInterface $em */
		$em = $this->getDoctrine()->getManager();
		$customer = $em->getRepository(Customer::class)->find(1);
		// $customer type is Customer|null|object
	}
}
```

**Comment by Jasonoro** [comment]
This seems fixed in PHPStorm 2020.2, @maldoinc is it fixed for you as well?

**Comment by maldoinc** [comment]
It does seem to be fixed. However the strangest bug appears where nested objects are no longer able to be autocompleted even though their type is known. Probably a phpstorm issue though.

Does that happen for you @Jasonoro ?

```php
$res = $this->getDoctrine()->getManager()->getRepository(Contact::class)->find(1);
$res->getCreatedAt()->format('d.m.Y');
```

In the above example the IDE is no longer able to autocomplete the DateTime methods from `getCreatedAt`. When type-hinting the `res` variable with a docblock everything works properly again.

**Comment by Jasonoro** [comment]
Ah sorry I missed your comment. The above works fine for me, however I've updated to PHPStorm 2020.2 in the mean time

**Comment by zhil** [comment]
Looks like I have this bug in PhpStorm 2020.2 with symfony plugin 0.21.203
I have entity Hospital, and SomeCommand.php

Sample code
```
        $hospitals = $this->em->getRepository(Hospital::class)->findBy(['oec_autoSchedule_enable' => true]);
        foreach ($hospitals as $hospital) {
            $hospitalDate = $hospital->getHospitalTimeDate();
        }
```

Hospital class is resolved, findBy method signature seems to be resolved (check gif), but $hospitals have type object[] :+1: 

![phpstorm](https://user-images.githubusercontent.com/981783/96886582-63f66080-148c-11eb-9239-2757aca1a2fc.gif)

Any ideas how can I fix it without adding 
```
/** @var $hospitalDate Hospital */
```
?

**Comment by zhil** [comment]
@mattsches @jensschulze @Haehnchen sorry for annoying, any ideas what could be reason for this bug? Is it PhpStorm bug OR plugin?

**Comment by syteknet-core** [comment]
this is no bug. it is just using the return type of the findBy method. you can create a separate method with a more specific return type or the typehint you posted.

**Comment by oxan** [comment]
@syteknet-core I don't think so. This is still reproducible if you annotate the findBy method with the correct return type, and PhpStorm does show the correct signature for the findBy method. Also, it works fine when the Symfony plugin isn't activated.

**Comment by zhil** [comment]
@syteknet-core also pretty sure it worked correctly with previous versions, not sure when does it stop working.

**Comment by syteknet-core** [comment]
sorry devs, surely i might be wrong but i have never seen this working when fetching the repository directly from the entity manager. Must add, i have seen this very rarely in recent projects.

You can also create a custom Service Entity Repository, add the correct annotation for the findBy method in the classes docblock.

**Comment by zhil** [comment]
@syteknet-core ok, thanks for explanation.

Just wonder - is it possible to fix this issue by customizing Php Symfony2 Plugin? I noticed it have a lot of settings, not sure I understand them :)

**Comment by web-monster** [comment]
```php
    /**
     * @var EntityManagerInterface|ObjectManager
     */
    private EntityManagerInterface $entityManager;
```

If you add **ObjectManager** (_Doctrine\Persistence\ObjectManager_) to the variable annotation the entity is defined (->findOneBy, ->findBy, ->find return App\Entity\User),  only the EntityManagerInterface does not define (return null|object)
